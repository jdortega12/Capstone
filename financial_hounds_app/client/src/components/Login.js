import React from "react";
import { useState } from 'react';
import "../styles/Login.css";
import axios from 'axios';


const Login = () => {

    const [username, setUsername] = useState("");
    const [password, setPassword] = useState("");

    const handleSubmit = async(e) => {
      const loginData = {"username": username, "password": password};
      alert(JSON.stringify(loginData));

      try{
        await axios({
          method: "post",
          url: "/login",
          data: loginData,
        });
      } catch(error){
        console.log(error)
      }
        
    };

    return (
      <div className="myDiv">
        <form className="myForm" method='POST'>
            <h3 className="myLabelLogin">Login</h3>
            <div className="mb-3">
                <label className="myLabelLogin">Username</label>
                <input
                    type="text"
                    value = {username}
                    onChange = {(e) => setUsername(e.target.value)}
                    className="form-control"
                    placeholder="Username"
                />
        </div>
        <div className="mb-3">
                <label className="myLabelLogin">Password</label>
                <input
                    type="password"
                    value = {password}
                    onChange = {(e) => setPassword(e.target.value)}
                    className="form-control"
                    placeholder="Username"
                />
        </div>
        <div className="d-grid">
          <button onClick={()=>handleSubmit()} className="myButton">
            Submit
          </button>
        </div>
        </form>
      </div>
    );
  }
  
export default Login;