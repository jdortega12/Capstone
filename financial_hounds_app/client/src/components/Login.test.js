import React from 'react';
import Login from '../components/Login';
import renderer from 'react-test-renderer';

test("Login Snapshot Test", ()=> {
    const component = renderer.create(
        <Login />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});