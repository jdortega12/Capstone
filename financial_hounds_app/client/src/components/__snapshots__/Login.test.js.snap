// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot Test 1`] = `
<div
  className="myDiv"
>
  <form
    className="myForm"
  >
    <h3
      className="myLabelLogin"
    >
      Login
    </h3>
    <div
      className="mb-3"
    >
      <label
        className="myLabelLogin"
      >
        Username
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="myLabelLogin"
      >
        Password
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Username"
        type="password"
        value=""
      />
    </div>
    <div
      className="d-grid"
    >
      <button
        className="myButton"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
